src/Microsoft.AspNetCore.Authentication.OpenIdConnect/Events/IOpenIdConnectEvents.cs

```diff
@@ -38,17 +28,22 @@ public interface IOpenIdConnectEvents : IRemoteAuthenticationEvents
         /// <summary>
         /// Invoked before redirecting to the identity provider to authenticate.
         /// </summary>
-        Task RedirectToAuthenticationEndpoint(RedirectContext context);
+        Task RedirectToIdentityProvider(RedirectContext context);
```

Comment:
We actually have abundant data from the WIF days, showing that most developers used the corresponding event without any deep knowledge of WS-Federation. The ability to modify the outgoing messages is a generic extensibility point, which has nothing to do with Azure AD - again, this event was in the WIF OM well before Azure AD even existed. _Modifying _the message does require more knowledge of the protocol, however _augmenting _the message  does not - it's common for people to add extra stuff to communicate to their own provider. And even modifying the message does not automatically entail _perfect_ knowledge of the protocol. A developer can have some familiarity with protocol artifacts that touch directly their scenarios (for example, scopes required by the API they talk to) but not with the protocol endpoints or flows.